### PR TITLE
Genetics rework part III: Stardust Crusaders

### PR DESCRIPTION
**Pull request by hex37**
## About The Pull Request
<details>
<summary>
	About The Pull Request
</summary>
<hr>

Adds the Genetic Analyzer, a complex machine that can identify, modify, manage mutations, then upload researched mutations directly to R&D. Comes with a completely custom Nano UI menu. This is the final machine for the core Genetics lab.

Adds Bear crates, available for purchase through the hydroponics menu in Lonestar. Order your bear today! (Plaid shirt not included)

Edited the description for the genetics Extractor.

Added UI Data functions to the backend of the Genetics framework

Made several genetics devices speak as if a particular scientist did the voice clips for them~

Buffed the Hand Labeler.

Adds the Genetic Analyzer, and bear crates to the centcomm testing area, along with a few other tools.

Adds a Cloning recipe to make Mukwah from bear DNA

Buffs SI

	
<hr>
</details>

## Changelog
:cl:
Adds the Genetic Analyzer, a complex machine that can identify, modify, manage mutations, then upload researched mutations directly to R&D. Comes with a completely custom Nano UI menu. This is the final machine for the core Genetics lab.

Adds Bear crates, available for purchase through the hydroponics menu in Lonestar. Order your bear today! (Plaid shirt not included)

Edited the description for the genetics Extractor.

Added UI Data functions to the backend of the Genetics framework

Made several genetics devices speak as if a particular scientist did the voice clips for them~

Buffed the Hand Labeler.

Adds the Genetic Analyzer, and bear crates to the centcomm testing area, along with a few other tools.

Adds a Cloning recipe to make Mukwah from bear DNA

Buffed SI
/:cl:

### CODE REVIEW

File: code/__DEFINES/genetics.dm [hunk 1]
```diff
@@ -10,4 +10,18 @@
...
+//These values are held by the flags variable in the /datum/genetics/mutation class.
+//They handle certain things happening dynamically when a mutation is added (IE: Destroying an abiotic part messily)
+#define MUT_FORCE_REQ 0x1 //If the target doesn't have requirements, it will force out abiotic parts.
+#define MUT_PURE_BIO_REQ 0x11 //Target needs to be entirely biotic and have no abiotic augments.
+#define MUT_FULL_BIO_REQ 0x12 //Target must only have biotic body parts (Not checking internals)
+#define MUT_BIO_ARM_REQ 0x13 //Target needs biotic arms
+#define MUT_BIO_LEG_REQ 0x14  //Target needs biotic legs
+#define MUT_BIO_UPPER_BODY_REQ 0x15  //Target needs biotic upper body
+#define MUT_BIO_LOWER_BODY_REQ 0x16  //Target needs biotic lower body
+#define MUT_BIO_HEAD_REQ 0x17 //Target needs a biotic head.
```
Comment (r4d6) [inline code/__DEFINES/genetics.dm:25]: Those don't look like the good values for bitflags.
Looking at other examples, it's 0x1, 0x2, 0x4, 0x8, 0x10, 0x20, 0x40, 0x80, 0x100, 0x200, 0x400, 0x800...

**Comment by Dopiotl** [comment]
Rapid gains

File: code/modules/genetics/recipes/mutation_recipe_defines.dm [hunk 2]
```diff
@@ -56,7 +56,7 @@
 					compare_string = compare_string + "M~"
 			else
 				compare_string = compare_string + "M~"
-
+		log_debug("initialize_mutation_recipes(): Loaded recipe [example_recipe] with string [compare_string]")
```
Comment (r4d6) [inline code/modules/genetics/recipes/mutation_recipe_defines.dm:59]: Just double-check that the log is supposed to stay.

**Review by r4d6 — COMMENTED** [review]
Only problem I see are the bit-flags being not defined correctly

File: code/__DEFINES/genetics.dm (hunk 1 above)
Comment (hex37) [reply code/__DEFINES/genetics.dm]: Fixed things to be powers of 2

File: code/modules/genetics/recipes/mutation_recipe_defines.dm (hunk 2 above)
Comment (hex37) [reply code/modules/genetics/recipes/mutation_recipe_defines.dm:59]: Yeah we want that there to make sure that recipe strings are loading like they ought to

**Pull request merged**